### PR TITLE
fix gemma3 grad acc

### LINKED ISSUE #37197: Gemma3 Gradient Accumulation loss
### System Info

transformers==4.50.3

### Who can help?

@osanseviero @ArthurZucker 

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

repro script with SFTTrainer here: https://gist.github.com/winglian/569924fe154824c8ce148f6e185cd4cd

When using `gradient_accumulation_steps=1`, the magnitude of the loss for gemma3 family of models is approximately correct ~1 or less depending on the model size. When changing `gradient_accumulation_steps=4`, the loss is scaled up by a factor of the `gradient_accumulation_steps` and in this case is ~8-10.

Digging into the [Gemma3TextModel.forward method ](https://github.com/huggingface/transformers/blob/800510c67bfc5cedd0bb7635648a07f39719be43/src/transformers/models/gemma3/modular_gemma3.py#L599), we see that it doesn't accept the typical `LossKwargs` to handle gradient accumulation. 

Compare the signature to [llama](https://github.com/huggingface/transformers/blob/800510c67bfc5cedd0bb7635648a07f39719be43/src/transformers/models/llama/modeling_llama.py#L809)  where it's defined as
```python
class KwargsForCausalLM(FlashAttentionKwargs, LossKwargs): ...
```

### Expected behavior

loss should be ~1.0 regardless of gradient accumulation steps.

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do?

This PR fixes the grad acc issue with gemma3 model. The issue was that we passed **kwargs in the model forward, so we were making the assumption that he was passing `**loss_kwargs` -> `num_items_in_batch` to calculate the loss. Not sure what is the best way to fix this @ArthurZucker in general as this might probably happen again. Maybe set `accepts_loss_kwargs` to `False` in general and set it to `True` for models that we fixed ? I'm fine also just setting it `False` for models that don't use the `kwargs` for the loss. 

As for why I didn't have the loss function: In the code, they are filetring the logits/labels so I decided to simply not use `num_items_in_batch` to calculate the loss. Otherwise, the loss won't be correctly calculated for one of the cases.

Also I fixed an issue related to peft as we couldn't have access to that attribute as the model was a peft model.

#### To reproduce
winglian script
https://gist.github.com/winglian/569924fe154824c8ce148f6e185cd4cd

### After fix 
grad acc 2 bs 1 and  grad acc 1 bs 2 
Screenshot 2025-04-02 at 4 38 54 PM

Fixes #37197

cc @winglian

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]
Nice job! Looks good to me, is `get_base_model` an old enough api we don’t need to worry about breaking stuff?

**Comment by SunMarc** [comment]
> Nice job! Looks good to me, is get_base_model an old enough api we don’t need to worry about breaking stuff?


Should be. we are already using it in the past here: 

```python
 model_forward = (
            unwrapped_model.forward
            if not _is_peft_model(unwrapped_model)
            else unwrapped_model.get_base_model().forward
        )
```

**Comment by SunMarc** [comment]
cc @ArthurZucker gentle ping

**Review by ArthurZucker — APPROVED** [review]
Thanks let's make it go green and merge

**Comment by ArthurZucker** [comment]
@bot /style

File: setup.py [hunk 1]
```diff
@@ -319,7 +319,6 @@ def run(self):
     "pyctcdecode",
     "phonemizer",
     "kenlm",
-)
 # `pip install ".[speech]"` is deprecated and `pip install ".[torch-speech]"` should be used instead
```
Comment (ArthurZucker) [inline setup.py]:
```suggestion
)
# `pip install ".[speech]"` is deprecated and `pip install ".[torch-speech]"` should be used instead
```

**Review by ArthurZucker — COMMENTED** [review]
oups sorry

**Comment by ArthurZucker** [comment]
@bot /style
